components/datetime/src/dynamic.rs

```diff
@@ -0,0 +1,456 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(feature = "serde")]
+use crate::neo_serde::*;
+use crate::options::preferences::HourCycle;
+use crate::raw::neo::RawNeoOptions;
+use crate::scaffold::GetField;
+use crate::{fields, fieldset, NeoSkeletonLength};
+use icu_provider::prelude::*;
+
+/// An enumeration over all possible date field sets.
+///
+/// 📏 Caution: This enumeration links more data than the
```

Comment:
nit: "links more data" when? just mentioning the enum clearly doesn't, but it does when used as a fieldset parameter to Formatter. I don't think people will assume that constructing the enum will link in additional data, but they won't necessarily know what the threshold is.